libs/mixer/melody.ts

```diff
@@ -110,6 +110,60 @@ namespace music {
 
     let playToneID = 0
 
+    /**
+     * Play a melody from the melody editor.
+     * @param melody
+     * @param tempo
+     */
+    //% block="play melody $melody at tempo $tempo|(bpm)" blockId=playMelodyFromEditor
+    //% blockNamespace=music
+    //% weight=75 blockGap=8
+    //% group="Melody"
+    //% melody.shadow="melody_editor"
+    //% tempo.min=40 tempo.max=500
+    //% tempo.defl=120
+    export function playMelodyFromEditor(melody: string, tempo: number) {
+        // add tempo to string so it is reflected in simulation
+        // creates format like "C5-174 B4 A G F E D C "
+        let formattedMelody = melody.substr(0, melody.indexOf(' ')) + "-" + tempo + melody.substr(melody.indexOf(' '), melody.length);
```

Comment:
Break up this expression, maybe move the indexOf result into a variable

Reply:
I rewrote it this since I moved the parsing into here and it's much easier to read now. :)